packages/@storefront/core/test/unit/core/tag/index.ts

```diff
@@ -171,4 +170,70 @@ suite('Tag', ({ expect, spy, stub }) => {
       expect(clazz[TAG_DESC]).to.eq(description);
     });
   });
+
+  describe('subscribeWith', () => {
+    it('should invoke `flux.all` with the events and callback', () => {
+      const events = ['a', 'b', 'c'];
+      const all: any = spy();
+      const cb = spy();
+      tag.flux = { all };
+      tag.one = () => null;
+
+      tag.subscribeWith(events, cb);
+
+      expect(all).to.be.calledWith(events, cb);
+    });
+
+    it('should update the _lookups array', () => {
+      const events = ['a', 'b', 'c'];
+      const cb = spy();
+      tag.flux = { all: () => null };
+      tag.one = () => null;
+
+      tag.subscribeWith(events, cb);
+
+      expect(tag._lookups).to.eql([[events, cb]]);
+    });
+
+    it('should register the unmount listener', () => {
+      tag.one = spy();
+      tag.flux = { all: () => null };
+      tag._removeLookups = spy();
+
+      tag.subscribeWith(['a', 'b', 'c'], () => null);
+
+      expect(tag.one).to.be.calledWith(Phase.UNMOUNT, tag._removeLookups);
+    });
+
+    it('should not re-register the unmount listener', () => {
+      tag.one = spy();
+      tag.flux = { all: () => null };
+      tag._lookups = [
+        [['a', 'b', 'c'], () => null]
+      ];
+
+      tag.subscribeWith(['d', 'e', 'f'], () => null);
+
+      expect(tag.one).to.not.be.called;
+    });
+  });
+
+  describe('_removeLookups()', () => {
+    it('should call allOff for each lookup', () => {
+       const events = ['a', 'b', 'c', 'd', 'e', 'f'];
+       const cb = () => null;
+       const cb2 = () => null;
+       const cb3 = () => null;
+       tag._lookups = [
+         [[events[0], events[1]], cb],
+         [[events[2]], cb2],
+         [[events[3], events[4], events[5]], cb3]
+       ];
+       tag.flux = { allOff: spy() };
+
+       tag._removeLookups();
+
+       expect(tag.flux.allOff.args).to.eql(tag._lookups);
+    });
+  });
```

Comment:
Is it worth testing the case for non-disjoint events?

Reply:
Discussed IRL: not relevant for proposed implementation. Marking as resolved.